sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -1032,6 +1039,75 @@ public void processElement(ProcessContext c) {
      }
   }
 
+  private static class MyInteger implements Comparable<MyInteger> {
+    private final int value;
+
+    MyInteger(int value) {
+      this.value = value;
+    }
+
+    public int getValue() {
+      return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+
+      if (!(o instanceof MyInteger)) {
+        return false;
+      }
+
+      MyInteger myInteger = (MyInteger) o;
+
+      return value == myInteger.value;
+
+    }
+
+    @Override
+    public int hashCode() {
+      return value;
+    }
+
+    @Override
+    public int compareTo(MyInteger o) {
+      return Integer.compare(this.getValue(), o.getValue());
+    }
+
+    @Override
+    public String toString() {
+      return "MyInteger{" + "value=" + value + '}';
+    }
+  }
+
+  private static class MyIntegerCoder extends CustomCoder<MyInteger> {
+    private static final MyIntegerCoder INSTANCE = new MyIntegerCoder();
+
+    public static MyIntegerCoder of() {
+      return INSTANCE;
+    }
+
+    @Override
+    public void encode(MyInteger value, OutputStream outStream, Context context)
+        throws CoderException, IOException {
+      ObjectOutputStream objectOutputStream = new ObjectOutputStream(outStream);
+      objectOutputStream.writeInt(value.getValue());
+      objectOutputStream.flush();
+    }
+
+    @Override
+    public MyInteger decode(InputStream inStream, Context context) throws CoderException,
+        IOException {
+      if (inStream.available() > 0) {
```

Comment:
A simpler way to make this foolproof might be to just internally delegate to a `VarIntCoder` anyhow. It would also make it more obvious that the implementation of the coder doesn't matter.